src/Jobs/ResizeImagesJob.php

```diff
@@ -18,15 +19,18 @@ class ResizeImagesJob implements ShouldQueue, ShouldBeUnique
     use Queueable;
 
     public function __construct(
+        public $forceAll = false
     ) {
         $this->onConnection(config('image-optimize.default_queue_connection'));
         $this->onQueue(config('image-optimize.default_queue_name'));
     }
 
-    public function handle(): void
+    public function handle(): Batch
     {
         $assets = Asset::all();
```

Comment:
Wat may be a prettier implementation (If supported by statamic)
```suggestion
        $assets = Asset::query()
            ->when(!$forceAll, fn ($q) => $q->whereNull('image-optimized'))
            ->get();
```

Reply:
Sadly, as stated in the comment above. It's not how the Statamic collection works..